File: ueueo-framework/ueueo-object-extending/src/main/java/com/ueueo/data/objectextending/modularity/ModuleExtensionConfiguration.cs
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Volo.Abp.ObjectExtending.Modularity;

public class ModuleExtensionConfiguration
{
    [NotNull]
    public EntityExtensionConfigurationDictionary Entities;//  { get; }

    [NotNull]
    public Dictionary<String, Object> Configuration;//  { get; }

    public ModuleExtensionConfiguration()
    {
        Entities = new EntityExtensionConfigurationDictionary();
        Configuration = new Dictionary<String, Object>();
    }
}

File: ueueo-framework/ueueo-object-extending/src/main/java/com/ueueo/data/objectextending/modularity/ModuleExtensionConfiguration.java
package com.ueueo.data.objectextending.modularity;

import lombok.Getter;
import org.springframework.lang.NonNull;

import java.util.HashMap;
import java.util.Map;
import java.util.function.Consumer;

@Getter
public class ModuleExtensionConfiguration {
    @NonNull
    private Map<String, EntityExtensionConfiguration> entities;

    @NonNull
    public Map<String, Object> configuration;

    public ModuleExtensionConfiguration() {
        entities = new HashMap<>();
        configuration = new HashMap<>();
    }

    public ModuleExtensionConfiguration configureEntity(String objectName, Consumer<EntityExtensionConfiguration> configureAction) {
        EntityExtensionConfiguration configuration = entities.computeIfAbsent(
                objectName,
                s -> new EntityExtensionConfiguration()
        );

        configureAction.accept(configuration);
        return this;
    }
}
